torch/_functorch/_aot_autograd/input_output_analysis.py

```diff
@@ -403,13 +405,63 @@ def compute_overlapping_inputs(fwd_inputs, aliased_input_indices):
                 err_message,
                 case_name="dynamic_shapes_validation",
             )
-    for j in range(num_aliases):
-        for i in range(j):
-            j_ = aliased_input_indices[j]
-            i_ = aliased_input_indices[i]
-            if not _tensors_definitely_do_not_overlap(fwd_inputs[i_], fwd_inputs[j_]):
-                actual_aliased_indices.add(i_)
-                actual_aliased_indices.add(j_)
+
+    shape_env = None
+    maybe_suppress_guards = contextlib.nullcontext
+    tracing_context = torch._guards.TracingContext.try_get()
+
+    if tracing_context is not None:
+        shape_env = tracing_context.fake_mode.shape_env
+
+        # Check whether we can actually get the dynamo sources from within AOTAutograd.
+        if aot_config.aot_autograd_arg_pos_to_source and shape_env is not None:
+            maybe_suppress_guards = shape_env.suppress_guards
+
+    # Check whether there are any symbolic values being used.
+    # We do this for 2 reasons:
+    #   1. StorageOverlap guard is only issued whenever dynamic shapes is turned on
+    #   2. Triggers the fast-path for computing storage overlapping
+    symbolic = any(
+        isinstance(x, torch.SymInt)
+        for i in aliased_input_indices
+        for x in [*fwd_inputs[i].shape, *fwd_inputs[i].stride(), fwd_inputs[i].storage_offset()]
+    )
```

Comment:
@bdhirsh Could you confirm this is the right check I'm doing, here? 
i.e. Only issue an `StorageOverlap` guard if  there's any `SymInt` in any of the aliased tensors.